### PR TITLE
Upgrade dev-setup versions / arm64 support

### PR DESCRIPTION
**Pull request by adracus**
* Upgrade dev-setup etcd version to 3.5.1
* Upgrade dev-setup kube-apiserver version to 1.22.1
* Verified Gardener runs on an M1 MacBook

/area dev-productivity

### CODE REVIEW

**Comment by gardener-robot-ci-1** [comment]
Thank you @adracus for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: hack/local-development/local-garden/run-kube-apiserver [hunk 1]
```diff
@@ -12,7 +12,7 @@ fi
 LABEL=${1:-local-garden}
 ACTIVATE_SEEDAUTHORIZER=${2:-false}
 
-IMAGE=k8s.gcr.io/kube-apiserver:v1.20.2
+IMAGE=k8s.gcr.io/kube-apiserver:v1.22.2
```
Comment (timebertt) [inline hack/local-development/local-garden/run-kube-apiserver]:
```suggestion
IMAGE=k8s.gcr.io/kube-apiserver:v1.22.3
```

**Review by timebertt — COMMENTED** [review]
Thank you!
Tested it on my machine (x86), didn't observe anything suspicious.

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**